src/components/Modal/_modals/TransactionDetailsModal/TransactionDetailsModal.tsx

```diff
@@ -44,50 +45,69 @@ export const TransactionDetailsModal: FC<TransactionDetailsProps> = ({ open, pri
         })
   }, [apiUrl, id])
 
+  const isIncoming = transactionData?.direction === 'incoming'
+  const isOutgoing = transactionData?.direction === 'outgoing'
+
   return (
     <Modal
       open={open}
       modalTitle="Transaction details"
-      primaryButton={{ text: 'Close', variant: 'reject', onClick: primaryButtonOnClickHandler }}
-      onCloseByEsc={primaryButtonOnClickHandler}
+      primaryButton={{ text: 'Close', variant: 'reject', onClick: onClose }}
+      onCloseByEsc={onClose}
     >
       <>
         {loading && <Loader />}
         {errors ? (
           <ErrorBar errorMsg={errors} withReloadButton />
+        ) : transactionData == null ? (
+          <ErrorBar errorMsg="Could not load transaction data" withReloadButton />
         ) : (
           <DetailsList>
+            <ListElement>
+              <DataName>Sender:</DataName>
+              <Highlighted enabled={isIncoming}>
+                {transactionData.sender}
+                {isIncoming && <SetPaymailButton paymail={transactionData.sender} onClick={onClose} />}
+              </Highlighted>
+            </ListElement>
+            <ListElement>
+              <DataName>Receiver:</DataName>
+              <Highlighted enabled={isOutgoing}>
+                {transactionData.receiver}
+                {isOutgoing && <SetPaymailButton paymail={transactionData.receiver} onClick={onClose} />}
+              </Highlighted>
+            </ListElement>
```

Comment:
I've added `sender` & `receiver` fields in the modal with transaction details.
![image](https://github.com/bitcoin-sv/spv-wallet-web-frontend/assets/152964795/8e07ec59-f43e-4b0f-9037-22fc9c7d5fe6)